orchestrator/crates/orchestrator/src/core/client/database/mongodb.rs

```
@@ -442,4 +449,110 @@ impl DatabaseClient for MongoDbClient {
+
+        match cursor.try_next().await? {
+            Some(doc) => {
+                // Try to deserialize and log any errors
+                match mongodb::bson::from_document::<Batch>(doc.clone()) {
+                    Ok(batch) => {
+                        let attributes = [KeyValue::new("db_operation_name", "get_latest_batch")];
+                        let duration = start.elapsed();
+                        ORCHESTRATOR_METRICS.db_calls_response_time.record(duration.as_secs_f64(), &attributes);
+                        Ok(Some(batch))
+                    }
+                    Err(e) => {
+                        tracing::error!(
+                            error = %e,
+                            document = ?doc,
+                            "Failed to deserialize document into Batch"
+                        );
+                        Err(DatabaseError::FailedToSerializeDocument(format!("Failed to deserialize document: {}", e)))
+                    }
+                }
+            }
+            None => Ok(None),
+        }
+    }
+
+    async fn update_batch(&self, batch: &Batch, update: &BatchUpdates) -> Result<Batch, DatabaseError> {
+        let start = Instant::now();
+        let filter = doc! {
+            "id": batch.id,
+        };
+        let options = FindOneAndUpdateOptions::builder().upsert(false).return_document(ReturnDocument::After).build();
+
+        let mut updates = update.to_document()?;
+
+        // remove null values from the updates
+        let mut non_null_updates = Document::new();
+        updates.iter_mut().for_each(|(k, v)| {
+            if v != &Bson::Null {
+                non_null_updates.insert(k, v);
+            }
+        });
+
+        // throw an error if there's no field to be updated
+        if non_null_updates.is_empty() {
+            return Err(DatabaseError::NoUpdateFound("No field to be updated, likely a false call".to_string()));
```

do we need to throw an error when there is nothing to update? how about we throw a warning and just return the batch?

---

This is the pattern we are following in update job as well. 
When I am calling this method in the code as well, I am assuming that the doc should already be present and I expect an error if it's not able to update it
If it's a hard requirement, can make the changes in the code. But I would prefer to keep it this way :)

---

@prkpndy , we might have legacy code in update job ignore that for now we can correct that in future scope,
To answer to the @Mohiiit comment, it is not madatory to suppress the error in all the case with the warning, 

My recommendation would be like create two function 
- update_item_if_exists - > this function will not throw any error if nothing to been updated return `Optional<Item>`
- update_item - will update or raise error if the object is not present in the system